### DISCUSSION TITLE
[Steps]: Going to a previous step marks a completed step as incomplete

### DISCUSSION THREAD

**Discussion by veloware** [chakra-ui/zag#2148] [General]
I was attempting to the Steps machine for the steps in a multi-step form and noticed that the steps logic is quite basic, in that it uses the current index compared to the list of steps to define which step is complete and which isn't.  All steps before the current step are complete, and all steps after the current step are marked as incomplete.

At first I  just thought 'oh well, the machine logic is not going to suit what I need', but when thought about it a little, i'm curious if this is an oversight.  I can't think of many scenarios of when in a multi-step process, that by navigating to  a previous step it should mark a previously completed step back to incomplete?

Is there an appetite to add logic for a scenario similar to the 'linear' prop to allow for a variation that when a step is complete, it cannot be incomplete?

Similar to MUI's stepper non-linear scenario - https://mui.com/material-ui/react-stepper/?srsltid=AfmBOooCI-aCRmWE4I6IMKbYpztDGayN2KGiMwOJ9iMOpZjCd7ptadLD#non-linear

side... an invalid state seems a reasonable consideration too?

**Comment by segunadebayo** [comment]
That sounds like a good idea. What do you have in mind in terms of the API or Code?
